### PR TITLE
Neptune Analytics Support & Logging

### PR DESCRIPTION
**Pull request by sophiadt**
- ## PR to release #19 changes to main

   - This PR is to support connection to Neptune Analytics and additional logging enhancements. Support for Neptune Analytics SDK has not been added yet.

### Changes:
- corrected the consoleOut.error(msg) statements to console.error(msg)
- added an if statement to only retrieve the database cluster information if the type is `neptune-db`
 - added in Andrea Child's changes from 42fa1017308a9ca3381b4ddb3ad924c97a71dd0d
    - fixed `todo.schema.graphql` that failed to create when utility is used with the todo schema
    - fixed errors in the tests cases ex. missing commas and incorrect test case ordering

Note: connecting to a Neptune Analytics graph requires the additional argument: `--create-update-aws-pipeline-neptune-IAM option`

### CODE REVIEW

File: src/NeptuneSchema.js [hunk 1]
```diff
@@ -52,46 +54,68 @@ function yellow(text) {
...
 
 function consoleOut(text) {
-    if (VERBOSE) {
-        console.log(text);
-    }
+    loggerLog(text, VERBOSE);
 }
 
+function sanitize(text) {
+    // TODO implement sanitization logic
```
Comment (Cole-Greer) [inline src/NeptuneSchema.js:61]: Will this be coming in a future PR?

File: src/NeptuneSchema.js [hunk 2]
```diff
@@ -125,35 +153,31 @@ async function getEdgesNames() {
...
     let response = await queryNeptune(query);
-    let result = response.results[0];
-    if (result !== undefined) {                    
-        direction.edge.directions.push({from:direction.from, to:direction.to});
-        consoleOut('  Found edge: ' + yellow(direction.edge.label) + '  direction: ' + yellow(direction.from) + ' -> ' + yellow(direction.to));
+    for (let result of response.results) {
+        for (let fromLabel of result.fromLabel) {
+            for (let toLabel of result.toLabel) {
+                edgeStructure.directions.push({from:fromLabel, to:toLabel});
+                consoleOut('  Found edge: ' + yellow(edgeStructure.label) + '  direction: ' + yellow(fromLabel) + ' -> ' + yellow(toLabel));
```
Comment (Cole-Greer) [inline src/NeptuneSchema.js:172]: Is the expectation that calls to `consoleOut` should always be paired with calls to `loggerLog`?

File: src/NeptuneSchema.js (hunk 1 above)
Comment (andreachild) [reply src/NeptuneSchema.js:57]: Yes - still need to talk with AWS about the logic.

**Comment by andreachild** [comment]
This PR was broken into several smaller ones:
* https://github.com/aws/amazon-neptune-for-graphql/pull/29
* https://github.com/aws/amazon-neptune-for-graphql/pull/31
* https://github.com/aws/amazon-neptune-for-graphql/pull/32
Closing this PR.

**Comment by xiazcy** [comment]
> This PR was broken into several smaller ones:
> 
> * [Change neptune queries to use parameters #29](https://github.com/aws/amazon-neptune-for-graphql/pull/29)
> * [Integration with Neptune Analytics #31](https://github.com/aws/amazon-neptune-for-graphql/pull/31)
> * [Increased logging to console and introduced file logging #32](https://github.com/aws/amazon-neptune-for-graphql/pull/32)
>   Closing this PR.

Sounds good. Thank you @andreachild!

**Pull request closed without merging**